### ISSUE TITLE
A Proxy for ClickHouse with routing rules

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#23206]
Clickhouse Bouncer - lightweight connection gateway/reverse-proxy.

Support both HTTP (HTTPS?) / TCP Clickhouse protocols.

Ability to connect to live nodes.
Ability to connect to replica-sets depending on user profile. (Imagine we have 20 shards * 3 replica. 10 Super fast replica & 10+10 slow. And we want to route real-time queries from USER1 to the fast replicas and slow analytical queries from USER2, USER3 route to the slow replica-sets).

Basically https://github.com/Vertamedia/chproxy can do it but only for HTTP/HTTPS.
(https://github.com/Vertamedia/chproxy  _May proxy requests to multiple distinct ClickHouse clusters depending on the input user. For instance, requests from appserver user may go to stats-raw cluster, while requests from reportserver user may go to stats-aggregate cluster._)

**Comment by UnamedRus** [comment]
>  Ability to connect to replica-sets depending on user profile. (Imagine we have 20 shards * 3 replica. 10 Super fast replica & 10+10 slow. And we want to route real-time queries from USER1 to the fast replicas and slow analytical queries from USER2, USER3 route to the slow replica-sets).

Probably it can be done on distributed table level?

**Comment by den-crane** [comment]
USER1 should be connected to the fast replica and use this fast replica as an initiator

**Comment by cangyin** [comment]
Is there any updates or progress for ClickHouse Bouncer proxy?
